### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.ToIntervalMod

### PR DESCRIPTION
**Pull request by LukasMias**
Only now realized that this is part of an ongoing project within mathlib3, see https://github.com/leanprover-community/mathlib/pull/17743. Might be best to wait.

---

- [x] depends on: #2150 [port missing theorems]

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2150
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2150~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Algebra/Order/ToIntervalMod.lean [hunk 1]
```diff
@@ -0,0 +1,817 @@
...
+theorem toIocDiv_add_right' (a : α) {b : α} (hb : 0 < b) (x y : α) :
+    toIocDiv a hb (x + y) = toIocDiv (a - y) hb x := by
+  rw [← sub_neg_eq_add, toIocDiv_sub', sub_eq_add_neg]
+#align to_Ioc_div_add_right' toIocDiv_add_right'
+
+theorem toIcoDiv_neg (a : α) {b : α} (hb : 0 < b) (x : α) :
+    toIcoDiv a hb (-x) = -(toIocDiv (-a) hb x + 1) := by
+  suffices toIcoDiv a hb (-x) = -toIocDiv (-(a + b)) hb x by
+    rwa [neg_add, ← sub_eq_add_neg, ← toIocDiv_add_right', toIocDiv_add_right] at this
+  rw [eq_neg_iff_eq_neg, eq_comm]
```
Comment (MichaelStollBayreuth) [inline Mathlib/Algebra/Order/ToIntervalMod.lean:360]: `eq_neg_iff_eq_neg` was removed by #17483. Replace by `rw [← neg_eq_iff_eq_neg]`.

Comment (Parcly-Taxel) [reply Mathlib/Algebra/Order/ToIntervalMod.lean:360]:
```suggestion
  rw [← neg_eq_iff_eq_neg, eq_comm]
```

**Comment by eric-wieser** [comment]
Hmm, it looks like there's not much worth rescuing from here after all; `set_option synthInstance.etaExperiment` seems to clear out all the hacks.

**Pull request closed without merging**